### DISCUSSION TITLE
Accessibility/talkback and Balloon

### DISCUSSION THREAD

**Discussion by rahulmcs** [skydoves/Balloon#270] [Q&A] [answered]
Is there anyway for the balloon to focus itself automatically when shown using the showAlignBottom() method when accessibility talkback is on? If i tap on the view when talkback is on then the view reads out the text correctly but it doesnt focus itself automatically.

**Comment by rahulmcs** [comment, marked as answer]
@skydoves ? Would you have any suggestions?

**Reply by skydoves** [reply]
Hello @rahulmcs, could you explain more details about the `doesn't focus automatically`?
Which actions are expected?

**Reply by rahulmcs** [reply]
Basically, once the ballon has been shown using e.g.showAlignBottom(), i would like the tooltip to get focussed and talkback to read the contents of the tooltip automatically. Is there any in-built mechanism within the library for this to happen? If not then I tried the following:

createTooltipError(...).apply{
showAlignBottom(...)
            val handler = Handler()
            handler.postDelayed({

                getContentVew().performAccessibilityAction(
                    AccessibilityNodeInfo.ACTION_ACCESSIBILITY_FOCUS,
                    null
                )
                getContentVew().sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_SELECTED)
            }, 500)

}
The above code manages to focus the balloon view for a while but the text inside is not read out. 
If i manually tap the balloon when talkback is on, then it gets focussed as well as reads the contents of the view as well.
@skydoves

**Reply by rahulmcs** [reply]
i finally got it to work by also setting getContentView().importantForAccessibility = IMPORTANT_FOR_ACCESSIBILITY_YES before executing the code above.

**Reply by skydoves** [reply]
Thank you @rahulmcs  for sharing your solution!
